internal/build/phases_test.go

```diff
@@ -88,19 +110,65 @@ func testPhases(t *testing.T, when spec.G, it spec.S) {
 			fakePhaseFactory := fakes.NewFakePhaseFactory()
 			expectedNetworkMode := "some-network-mode"
 
-			err := lifecycle.Create(context.Background(), false, false, "test", "test", "test", "test", expectedNetworkMode, fakePhaseFactory)
+			err := lifecycle.Create(
+				context.Background(),
+				false,
+				false,
+				"test",
+				"test",
+				"test",
+				"test",
+				expectedNetworkMode,
+				[]string{},
+				fakePhaseFactory,
+			)
 			h.AssertNil(t, err)
 
 			configProvider := fakePhaseFactory.NewCalledWithProvider
 			h.AssertEq(t, configProvider.HostConfig().NetworkMode, container.NetworkMode(expectedNetworkMode))
 		})
 
+		it("configures the phase with both cache binds and custom volume mounts", func() {
```

Comment:
Since you added this check, maybe we can remove the tests on lines 212 and 332

Reply:
They still test slightly different circumstances - the difference between publish and not, as well as ensuring the cache binds will be configured regardless of whether a custom volume is configured.

Reply:
You're right but I'm not sure if I would keep all 3 tests. I think it's a little confusing that we're testing part of it 3 times.
Maybe we change this test to test only the configuration with custom volume mounts, and to keep the other tests to test only the cache and launch-cache.
What do you think about that?

Reply:
I opted to remove this test, and add custom volume mounts to the existing bind tests which cover both the `publish=true` and `publish=false` contexts.